tests/Microsoft.Identity.Test.Unit/TelemetryTests/HttpTelemetryTests.cs

```diff
@@ -390,6 +361,87 @@ await pca.AcquireTokenByUsernamePassword(TestConstants.s_scope, "username", Test
             }
         }
 
+        [TestMethod]
+        public async Task CallerSdkDetailsTestAsync()
+        {
+            using (_harness = CreateTestHarness())
+            {
+                _harness.HttpManager.AddInstanceDiscoveryMockHandler();
+                var requestHandler = _harness.HttpManager.AddMockHandlerSuccessfulClientCredentialTokenResponseMessage();
+
+                var cca = CreateConfidentialClientApp();
+
+                await cca.AcquireTokenForClient(TestConstants.s_scope)
+                    .WithExtraQueryParameters(new Dictionary<string, string> { 
+                        { "caller-sdk-id", "testApiId" },
+                        { "caller-sdk-ver", "testSdkVersion"} })
+                    .ExecuteAsync().ConfigureAwait(false);
+
+                AssertCurrentTelemetry(
+                    requestHandler.ActualRequestMessage, 
+                    ApiIds.AcquireTokenForClient, 
+                    CacheRefreshReason.NoCachedAccessToken,
+                    callerSdkId: "testApiId", 
+                    callerSdkVersion: "testSdkVersion");
+            }
+        }
+
+        [TestMethod]
+        public async Task CallerSdkDetails_ConstraintsTestAsync()
+        {
+            string callerSdkId = "testApiIdMoreThan10Chars";
+            string callerSdkVersion = "testSdkVersionMoreThan20Chars";
+
+            using (_harness = CreateTestHarness())
+            {
+                _harness.HttpManager.AddInstanceDiscoveryMockHandler();
+                var requestHandler = _harness.HttpManager.AddMockHandlerSuccessfulClientCredentialTokenResponseMessage();
+
+                var cca = CreateConfidentialClientApp();
+
+                await cca.AcquireTokenForClient(TestConstants.s_scope)
+                    .WithExtraQueryParameters(new Dictionary<string, string> {
+                        { "caller-sdk-id", callerSdkId },
+                        { "caller-sdk-ver", callerSdkVersion } })
+                    .ExecuteAsync().ConfigureAwait(false);
+
+                AssertCurrentTelemetry(
+                    requestHandler.ActualRequestMessage,
+                    ApiIds.AcquireTokenForClient,
+                    CacheRefreshReason.NoCachedAccessToken,
+                    callerSdkId: callerSdkId.Substring(0, Math.Min(callerSdkId.Length, 10)),

```

Comment:
I would move these numbers 10 and 20 to constants and add a meaningful name to better understand the test